include/deal.II/non_matching/mapping_info.h

```diff
@@ -0,0 +1,209 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2011 - 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef dealii_non_matching_mapping_info_h
+#define dealii_non_matching_mapping_info_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/aligned_vector.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/vectorization.h>
+
+#include <deal.II/fe/fe.h>
+#include <deal.II/fe/fe_update_flags.h>
+#include <deal.II/fe/mapping_cartesian.h>
+#include <deal.II/fe/mapping_q.h>
+
+#include <memory>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace NonMatching
+{
+  /**
+   * This class provides the mapping information computation and mapping data
+   * storage to be used together with FEPointEvaluation.
+   */
+  template <int dim, int spacedim = dim>
+  class MappingInfo
+  {
+  public:
+    /**
+     * Constructor.
+     *
+     * @param mapping The Mapping class describing the actual geometry of a cell.
+     *
+     * @param update_flags Specify the quantities to be computed by the mapping
+     * during the call of reinit().
+     */
+    MappingInfo(const Mapping<dim> &mapping, const UpdateFlags update_flags);
+
+    /**
+     * Reinitialize the mapping information for the incoming cell and unit
+     * points.
+     */
+    void
+    reinit(const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+           const ArrayView<const Point<dim>> &unit_points);
+
+    /**
+     * Getter function for current unit points.
+     */
+    const std::vector<Point<dim>> &
+    get_unit_points() const;
+
+    /**
+     * Getter function for computed mapping data.
```

Comment:
Do we want to write that this accesses internal data and is therefore not a stable interface?